Test project build fails when packages restored to inside project directory

## Description

Running `dotnet test` or just `dotnet build` with nuget packages restored inside the project directory results in this warning:

```
CSC : warning CS2002: Source file 'packages/microsoft.net.test.sdk/16.5.0/build/netcoreapp2.1/Microsoft.NET.Test.Sdk.Program.cs' specified multiple times [/home/test/Foo/Foo.csproj]
```

Attempting to build or test without the `--no-restore` option, this error happens instead:

```
/root/.nuget/packages/microsoft.net.test.sdk/16.7.1/build/netcoreapp2.1/Microsoft.NET.Test.Sdk.Program.cs(4,7): error CS0101: The namespace '<global namespace>' already contains a definition for 'AutoGeneratedProgram' [/Foo/Foo.csproj]
/root/.nuget/packages/microsoft.net.test.sdk/16.7.1/build/netcoreapp2.1/Microsoft.NET.Test.Sdk.Program.cs(3,2): error CS0579: Duplicate 'Microsoft.VisualStudio.TestPlatform.TestSDKAutoGeneratedCode' attribute [/Foo/Foo.csproj]
/root/.nuget/packages/microsoft.net.test.sdk/16.7.1/build/netcoreapp2.1/Microsoft.NET.Test.Sdk.Program.cs(4,41): error CS0111: Type 'AutoGeneratedProgram' already defines a member called 'Main' with the same parameter types [/Foo/Foo.csproj]
```

This only happens when the nuget `--package` option is pointed at the project directory itself, not the parent directory.

## Steps to reproduce

To get the warning:
```
dotnet new xunit -n Foo --no-restore; cd Foo; dotnet restore --packages packages; dotnet test --no-restore --diag:log.txt
```

To get the error:
```
dotnet new xunit -n Foo --no-restore; cd Foo; dotnet restore --packages packages; dotnet test --diag:log.txt
```

Tested with xunit and nunit.

## Expected behavior

Build / test command should succeed without warning or error.

## Actual behavior

Build and test commands fail / a warning is printed.

## Diagnostic logs

[log.txt](https://github.com/microsoft/vstest/files/5857352/log.txt)

It seems the generated `Program.cs` file appears on the command line twice, once with a relative and once with an absolute path? When logging with `--verbosity normal`:
```
CoreCompile:
         /usr/share/dotnet/dotnet exec "/usr/share/dotnet/sdk/5.0.102/Roslyn/bincore/csc.dll"
...
/analyzer:/usr/share/dotnet/sdk/5.0.102/Sdks/Microsoft.NET.Sdk/targets/../analyzers/Microsoft.CodeAnalysis.NetAnalyzers.dll /analyzer:/Foo/packages/xunit.analyzers/0.10.0/analyzers/dotnet/cs/xunit.analyzers.dll packages/microsoft.net.test.sdk/16.7.1/build/netcoreapp2.1/Microsoft.NET.Test.Sdk.Program.cs UnitTest1.cs /Foo/packages/microsoft.net.test.sdk/16.7.1/build/netcoreapp2.1/Microsoft.NET.Test.Sdk.Program.cs
```

## Environment

Tested on (unmodified) `mcr.microsoft.com/dotnet/core/sdk:3.1` and `mcr.microsoft.com/dotnet/sdk:5.0` docker images.
Also tested on local windows 10, `dotnet --info`:
```
.NET SDK (reflecting any global.json):
 Version:   5.0.102
 Commit:    71365b4d42

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.102\

Host (useful for support):
  Version: 5.0.2
  Commit:  cb5f173b96

.NET SDKs installed:
  1.0.0-preview2-1-003177 [C:\Program Files\dotnet\sdk]
  2.1.301 [C:\Program Files\dotnet\sdk]
  2.1.520 [C:\Program Files\dotnet\sdk]
  2.2.203 [C:\Program Files\dotnet\sdk]
  3.1.111 [C:\Program Files\dotnet\sdk]
  3.1.202 [C:\Program Files\dotnet\sdk]
  3.1.300 [C:\Program Files\dotnet\sdk]
  3.1.405 [C:\Program Files\dotnet\sdk]
  5.0.102 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.24 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.24 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.24 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

--- a/src/package/nuspec/netcoreapp/Microsoft.NET.Test.Sdk.targets
+++ b/src/package/nuspec/netcoreapp/Microsoft.NET.Test.Sdk.targets
@@ -35,6 +35,8 @@
           Condition="'$(GenerateProgramFile)' == 'true'">
 
     <ItemGroup Condition="'$(Language)' == 'VB' or '$(Language)' == 'C#'">
+      <RemoveExistingMicrosoftNETTestSdkProgram Include="@(Compile)" Condition="'%(FileName)' == 'Microsoft.NET.Test.Sdk.Program'" />
+			<Compile Remove="@(RemoveExistingMicrosoftNETTestSdkProgram)" />
       <Compile Include="$(GeneratedProgramFile)"/>
     </ItemGroup>
 

--- a/src/package/nuspec/netcoreapp/Microsoft.NET.Test.Sdk.targets
+++ b/src/package/nuspec/netcoreapp/Microsoft.NET.Test.Sdk.targets
@@ -36,7 +36,7 @@
 
     <ItemGroup Condition="'$(Language)' == 'VB' or '$(Language)' == 'C#'">
       <RemoveExistingMicrosoftNETTestSdkProgram Include="@(Compile)" Condition="'%(FileName)' == 'Microsoft.NET.Test.Sdk.Program'" />
-			<Compile Remove="@(RemoveExistingMicrosoftNETTestSdkProgram)" />
+      <Compile Remove="@(RemoveExistingMicrosoftNETTestSdkProgram)" />
       <Compile Include="$(GeneratedProgramFile)"/>
     </ItemGroup>
 

